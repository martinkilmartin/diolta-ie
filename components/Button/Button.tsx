type Props = {
  text: string
}

const Button = ({ text }: Props): JSX.Element => (
  <button className="btn">{text}</button>
)

export default Button
