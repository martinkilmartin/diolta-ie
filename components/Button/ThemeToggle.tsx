import { useState } from 'react'

type Props = {
  a?: string
  b?: string
}

const ThemeToggle = ({ a = 'dark', b = 'light' }: Props): JSX.Element => {
  const [theme, setTheme] = useState(true)
  const toggleTheme = () => setTheme((theme) => !theme)
  return (
    <button
      data-toggle-theme={`${a},${b}`}
      data-act-class="ACTIVECLASS"
      onClick={toggleTheme}
    >
      {theme ? '🌞' : '🌚'}
    </button>
  )
}

export default ThemeToggle
