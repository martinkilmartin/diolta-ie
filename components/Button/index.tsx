export { default as Button } from './Button'
export { default as ThemeToggle } from './ThemeToggle'
