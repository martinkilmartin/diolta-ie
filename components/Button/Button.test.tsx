import renderer from 'react-test-renderer'
import Button from './Button'

test('Button displays text', () => {
  const button = renderer.create(<Button text="Push Me" />)
  const tree = button.toJSON()
  expect(tree).toMatchSnapshot()
})
