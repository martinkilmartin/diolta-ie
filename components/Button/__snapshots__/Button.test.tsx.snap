// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button displays text 1`] = `
<button
  className="btn"
>
  Push Me
</button>
`;
